ktor-server/ktor-server-plugins/ktor-server-single-page/jvm/src/io/ktor/plugins/SinglePageApplication.kt

```diff
@@ -0,0 +1,169 @@
+        }
+
+        /**
+         * Creates an application configuration for the React project.
+         * Resources will be shared from the filesPath directory. The root file is index.html
+         */
+        public fun react(filesPath: String) {
+            this.filesPath = filesPath
+        }
+
+        /**
+         * Creates an application configuration for the Vue project.
+         * Resources will be shared from the filesPath directory. The root file is index.html
+         */
+        public fun vue(filesPath: String) {
+            this.filesPath = filesPath
+        }
+
+        /**
+         * Creates an application configuration for the Ember project.
+         * Resources will be shared from the filesPath directory. The root file is index.html
+         */
+        public fun ember(filesPath: String) {
+            this.filesPath = filesPath
+        }
+
+        /**
+         * Creates an application configuration for the Backbone project.
+         * Resources will be shared from the filesPath directory. The root file is index.html
+         */
+        public fun backbone(filesPath: String) {
+            this.filesPath = filesPath
+        }
+    }
+
+    /**
+     * The [SinglePageApplication] plugin implementation
+     */
+    public companion object Plugin : ApplicationPlugin<Application, Configuration, SinglePageApplication> {
+        override val key: AttributeKey<SinglePageApplication> = AttributeKey("SinglePage")
+
+        override fun install(pipeline: Application, configure: Configuration.() -> Unit): SinglePageApplication {
+            val plugin = SinglePageApplication(Configuration().apply(configure))
+
+            pipeline.routing {
+                static(plugin.applicationRoute) {
+                    if (plugin.usePackageNames) {
+                        resources(plugin.filesPath)
+                        defaultResource(plugin.defaultPage, plugin.filesPath)
+                    } else {
+                        staticRootFolder = File(plugin.filesPath)
+                        files(".")
+                        default(plugin.defaultPage)
+                    }
+                }
+            }
+
+            pipeline.sendPipeline.intercept(ApplicationSendPipeline.Before) {
+                plugin.interceptCall(this)
+            }
+            return plugin
+        }
+    }
+
+    private suspend fun interceptCall(
+        context: PipelineContext<Any, ApplicationCall>,
+    ) = with(context) context@{
+        val call = context.call
+        val requestUrl = call.request.uri
+
+        if (call.attributes.contains(key)) return@context
+
+        if (!isUriStartWith(requestUrl)) return@context
+
+        call.attributes.put(key, this@SinglePageApplication)
+
+        if (ignoredFiles.firstOrNull { it.invoke(requestUrl) } != null) {
+            call.response.status(HttpStatusCode.NotFound)
+            call.respondText("File with path $requestUrl is in ignore list")
```

Comment:
Not sure about the message. Maybe it can be configurable or remove it at all?

Reply:
I've removed this message